### PR TITLE
nil ptr check for endpointsettings when used with older clients

### LINKED ISSUE #19427: nil pointer on bridge connect in 1.10.0-rc1
Docker error log:

```
Jan 18 15:40:11 compooter docker[7858]: 2016/01/18 15:40:11 http: panic serving @: runtime error: invalid memory address or nil pointer dereference
Jan 18 15:40:11 compooter docker[7858]: goroutine 749 [running]:
Jan 18 15:40:11 compooter docker[7858]: net/http.(*conn).serve.func1(0xc821748000, 0x7f7520b46b58, 0xc820ffc000)
Jan 18 15:40:11 compooter docker[7858]: /usr/local/go/src/net/http/server.go:1287 +0xb5
Jan 18 15:40:11 compooter docker[7858]: github.com/docker/docker/daemon.(*Daemon).connectToNetwork(0xc820338420, 0xc820cf21c0, 0x16e63d8, 0x6, 0x0, 0x1, 0x0, 0x0)
Jan 18 15:40:11 compooter docker[7858]: /go/src/github.com/docker/docker/daemon/container_operations_unix.go:764 +0x1c9
Jan 18 15:40:11 compooter docker[7858]: github.com/docker/docker/daemon.(*Daemon).ConnectToNetwork(0xc820338420, 0xc820cf21c0, 0x16e63d8, 0x6, 0x0, 0x0, 0x0)
Jan 18 15:40:11 compooter docker[7858]: /go/src/github.com/docker/docker/daemon/container_operations_unix.go:741 +0x3ed
Jan 18 15:40:11 compooter docker[7858]: github.com/docker/docker/daemon.(*Daemon).ConnectContainerToNetwork(0xc820338420, 0xc820d58860, 0x20, 0x16e63d8, 0x6, 0x0, 0x0, 0x0)
Jan 18 15:40:11 compooter docker[7858]: /go/src/github.com/docker/docker/daemon/network.go:161 +0xb0
Jan 18 15:40:11 compooter docker[7858]: github.com/docker/docker/api/server/router/network.(*networkRouter).postNetworkConnect(0xc820eb1830, 0x7f75233bda70, 0xc820c2c630, 0x7f7520b5d1a8, 0xc821748210
Jan 18 15:40:11 compooter docker[7858]: /go/src/github.com/docker/docker/api/server/router/network/network_routes.go:125 +0x359
Jan 18 15:40:11 compooter docker[7858]: github.com/docker/docker/api/server/router/network.(*networkRouter).(github.com/docker/docker/api/server/router/network.postNetworkConnect)-fm(0x7f75233bda70, 
Jan 18 15:40:11 compooter docker[7858]: /go/src/github.com/docker/docker/api/server/router/network/network.go:40 +0x74
Jan 18 15:40:11 compooter docker[7858]: github.com/docker/docker/api/server.versionMiddleware.func1(0x7f75233bda70, 0xc820c2c630, 0x7f7520b5d1a8, 0xc821748210, 0xc82066a2a0, 0xc820c2c570, 0x0, 0x0)
Jan 18 15:40:11 compooter docker[7858]: /go/src/github.com/docker/docker/api/server/middleware.go:155 +0x83a
Jan 18 15:40:11 compooter docker[7858]: github.com/docker/docker/api/server.(*Server).corsMiddleware.func1(0x7f7520b46568, 0xc82000dab0, 0x7f7520b5d1a8, 0xc821748210, 0xc82066a2a0, 0xc820c2c570, 0x0,
Jan 18 15:40:11 compooter docker[7858]: /go/src/github.com/docker/docker/api/server/middleware.go:134 +0xfa
Jan 18 15:40:11 compooter docker[7858]: github.com/docker/docker/api/server.(*Server).userAgentMiddleware.func1(0x7f7520b46568, 0xc82000dab0, 0x7f7520b5d1a8, 0xc821748210, 0xc82066a2a0, 0xc820c2c570,
Jan 18 15:40:11 compooter docker[7858]: /go/src/github.com/docker/docker/api/server/middleware.go:117 +0x4be
Jan 18 15:40:11 compooter docker[7858]: github.com/docker/docker/api/server.(*Server).makeHTTPHandler.func1(0x7f7520b5d1a8, 0xc821748210, 0xc82066a2a0)
Jan 18 15:40:11 compooter docker[7858]: /go/src/github.com/docker/docker/api/server/server.go:169 +0x2ad
Jan 18 15:40:11 compooter docker[7858]: net/http.HandlerFunc.ServeHTTP(0xc820667a80, 0x7f7520b5d1a8, 0xc821748210, 0xc82066a2a0)
Jan 18 15:40:11 compooter docker[7858]: /usr/local/go/src/net/http/server.go:1422 +0x3a
Jan 18 15:40:11 compooter docker[7858]: github.com/gorilla/mux.(*Router).ServeHTTP(0xc820da8960, 0x7f7520b5d1a8, 0xc821748210, 0xc82066a2a0)
Jan 18 15:40:11 compooter docker[7858]: /go/src/github.com/docker/docker/vendor/src/github.com/gorilla/mux/mux.go:98 +0x29e
Jan 18 15:40:11 compooter docker[7858]: github.com/docker/docker/api/server.(*routerSwapper).ServeHTTP(0xc821193da0, 0x7f7520b5d1a8, 0xc821748210, 0xc82066a2a0)
Jan 18 15:40:11 compooter docker[7858]: /go/src/github.com/docker/docker/api/server/router_swapper.go:29 +0x72
Jan 18 15:40:11 compooter docker[7858]: net/http.serverHandler.ServeHTTP(0xc82000ad20, 0x7f7520b5d1a8, 0xc821748210, 0xc82066a2a0)
Jan 18 15:40:11 compooter docker[7858]: /usr/local/go/src/net/http/server.go:1862 +0x19e
Jan 18 15:40:11 compooter docker[7858]: net/http.(*conn).serve(0xc821748000)
Jan 18 15:40:11 compooter docker[7858]: /usr/local/go/src/net/http/server.go:1361 +0xbee
Jan 18 15:40:11 compooter docker[7858]: created by net/http.(*Server).Serve
Jan 18 15:40:11 compooter docker[7858]: /usr/local/go/src/net/http/server.go:1910 +0x3f6
```

Script to reproduce:

```
#!/bin/sh
docker network create custom || true
docker run --name repro -d busybox sleep 10000
go run main.go
```

main.go:
```
package main

import (
	"fmt"

	"github.com/samalba/dockerclient"
)

func main() {
	docker, err := dockerclient.NewDockerClient("unix:///var/run/docker.sock", nil)
	if err != nil {
		fmt.Println(err)
		return
	}
	config := &dockerclient.NetworkCreate{
		Name:           "custom",
		Driver:         "bridge",
		CheckDuplicate: true,
	}
	_, err = docker.CreateNetwork(config)
	if err != nil {
		if err.Error() == fmt.Sprintf("500 Internal Server Error: network with name %s already exists\n", "custom") {
			// that's okay because it already exists
			fmt.Println("network exists")
		} else {
			fmt.Printf("Failed to create Network %v\n", err)
			return
		}
	} else {
		fmt.Println("network created")
	}
	if err := docker.ConnectNetwork("bridge", "repro"); err != nil {
		fmt.Println(err)
		return
	}
}
```

### PR DESCRIPTION
**Pull request by mavenugo**
fixes https://github.com/docker/docker/issues/19427
Signed-off-by: Madhu Venugopal madhu@docker.com

### CODE REVIEW

File: daemon/container_operations_unix.go [hunk 1]
```diff
@@ -772,6 +768,9 @@ func (daemon *Daemon) connectToNetwork(container *container.Container, idOrName
 	}
 
 	if endpointConfig != nil {
+		if !containertypes.NetworkMode(idOrName).IsUserDefined() && len(endpointConfig.Aliases) > 0 {
+			return runconfig.ErrUnsupportedNetworkAndAlias
+		}
```
Comment (coolljt0725) [inline daemon/container_operations_unix.go]: I think it's better to also move `if err := validateNetworkingConfig(n, endpointConfig); err != nil {` here

Comment (mavenugo) [inline daemon/container_operations_unix.go]: though it is not required, since these functions already do this check. But am fine making this change.

**Comment by coolljt0725** [comment]
LGTM

**Comment by runcom** [comment]
LGTM

**Pull request merged**